### PR TITLE
[OTE_SDK_TEST_SUITE] Changes due adding deep-object-reid training tests

### PR DESCRIPTION
**Pull request by pfinashx**
Changes due adding Real-life training tests for Deep-Object_REID.
Please see job 425 - EfficinetNet-B0 on LG-chem dataset.
This PR depends on https://github.com/openvinotoolkit/deep-object-reid/pull/131 PR in Deep-Object_REID.
Changed KEEP_CONFIG_FIELD_VALUE, need update mmdetection real-life tests (in progress)

### CODE REVIEW

File: ote_sdk/ote_sdk/test_suite/training_tests_actions.py [hunk 1]
```diff
@@ -310,6 +309,16 @@ def create_openvino_task(model_template, environment):
     return openvino_task
 
 
+def get_task_class(path):
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/test_suite/training_tests_actions.py]: copy (not move!) https://github.com/openvinotoolkit/training_extensions/blob/develop/ote_cli/ote_cli/utils/importing.py to ote_sdk/ote_sdk/ and use function from there.

File: ote_sdk/ote_sdk/test_suite/training_tests_actions.py [hunk 2]
```diff
@@ -480,7 +491,7 @@ def _run_ote_nncf(
 
         logger.info("Run NNCF optimization")
         self.nncf_task.optimize(
-            OptimizationType.NNCF, dataset, self.nncf_model, OptimizationParameters()
+        OptimizationType.NNCF, dataset, self.nncf_model, None
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/test_suite/training_tests_actions.py]: indent

File: ote_sdk/ote_sdk/test_suite/training_tests_actions.py [hunk 3]
```diff
@@ -390,8 +399,10 @@ def _run_ote_pot(
         assert (
             self.optimized_model_pot.model_format == ModelFormat.OPENVINO
         ), "Wrong model format after pot"
+        #TO DO: Model.OptimizationType is "int enum" class, and OptimizationType is "enum",
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/test_suite/training_tests_actions.py]: let's way for a fix, don't fix it this way

**Review by Ilya-Krylov — CHANGES_REQUESTED** [review]

File: ote_sdk/ote_sdk/test_suite/training_tests_actions.py (hunk 3 above)
Comment (pfinashx) [reply ote_sdk/ote_sdk/test_suite/training_tests_actions.py]: Changed, already fixed in https://github.com/openvinotoolkit/deep-object-reid/pull/132

File: ote_sdk/ote_sdk/test_suite/training_tests_actions.py (hunk 2 above)
Comment (pfinashx) [reply ote_sdk/ote_sdk/test_suite/training_tests_actions.py]: Done!

File: ote_sdk/ote_sdk/test_suite/training_tests_actions.py (hunk 1 above)
Comment (pfinashx) [reply ote_sdk/ote_sdk/test_suite/training_tests_actions.py]: Done!

File: ote_sdk/ote_sdk/utils/importing.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+def get_impl_class(impl_path):
+    """Returns a class by its path in package."""
+
+    task_impl_module_name, task_impl_class_name = impl_path.rsplit(".", 1)
+    task_impl_module = importlib.import_module(task_impl_module_name)
+    task_impl_class = getattr(task_impl_module, task_impl_class_name)
+
+    return task_impl_class
+
+def get_task_class(path):
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/utils/importing.py]: hmm, what is the difference between `get_impl_class` and `get_task_class` ?

File: ote_sdk/ote_sdk/utils/importing.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    module_name, class_name = path.rsplit('.', 1)
+    module = importlib.import_module(module_name)
+    return getattr(module, class_name)
+
+def is_nncf_enabled():
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/utils/importing.py]: no, please don't bring nncf here

Comment (pfinashx) [reply ote_sdk/ote_sdk/utils/importing.py]: Changed

File: ote_sdk/ote_sdk/utils/importing.py (hunk 4 above)
Comment (pfinashx) [reply ote_sdk/ote_sdk/utils/importing.py]: Just little misunderstanding, fixed

**Comment by Ilya-Krylov** [comment]
run ote_sdk tests

**Comment by pfinashx** [comment]
run ote_sdk tests

**Comment by pfinashx** [comment]
run ote_sdk tests

**Comment by Ilya-Krylov** [comment]
20:29:40  [INFO] This may take a few minutes...
20:29:49  isort (ote_sdk)..........................................................Failed
20:29:49  - hook id: isort
20:29:49  - files were modified by this hook
20:29:49  
20:29:49  Fixing /home/jenkins/b/workspace/ote/precommit/ote_sdk_tests_runner/src/ote_sdk/ote_sdk/test_suite/training_tests_actions.py
20:29:49  
20:29:49  isort (ote_cli|external).................................................Passed
20:29:51  black (ote_sdk|ote_cli)..................................................Failed
20:29:51  - hook id: black
20:29:51  - files were modified by this hook
20:29:51  
20:29:51  reformatted ote_sdk/ote_sdk/test_suite/training_tests_actions.py
20:29:51  All done! ✨ 🍰 ✨
20:29:51  1 file reformatted, 188 files left unchanged.
20:29:51  
20:29:51  black (rest).............................................................Passed
20:29:52  flake8...................................................................Passed
20:29:53  prettier.................................................................Passed
20:29:57  mypy (ote_sdk)...........................................................Passed
20:29:57  mypy (ote_cli)...........................................................Passed
20:29:58  mypy (external)..........................................................Passed
20:30:30  pylint...................................................................Passed

**Comment by pfinashx** [comment]
run ote_sdk tests

**Review by Ilya-Krylov — APPROVED** [review]

**Pull request merged**